### ISSUE TITLE
Directives do not accept a negative numeric value

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#180]
Directive taking a negative  numeric value are not correctly parsed. For example: 

```
process foo {
  maxErrors -1 
  '''
  echo Hello
  '''
}
```

The `maxErrors` directive is *NOT* set to `-1`. What happens is that the statement is parsed as a binary expression i.e. `maxErrors` variable minus the value `1`, instead of setting the directive through a method invocation as expected. 

Workaround: wrap the number with parenthesis i.e. `maxErrors (-1)`  

See also https://github.com/nextflow-io/nextflow/issues/179#issuecomment-225177406

**Comment by pditommaso** [comment]
This has been solved with version `0.20.1`

**Comment by pditommaso** [comment]
It's worth noting that *negative* expression still needs to be enclosed with parenthesis. For example: 
```
process foo {
  maxErrors ( -1 + 2 * x )
  '''
  echo Hello
  '''
} 
```
